pkgs/applications/window-managers/sway/default.nix

```diff
@@ -21,6 +21,10 @@ stdenv.mkDerivation rec {
     ./load-configuration-from-etc.patch
   ];
 
+  postPatch = ''
+    substituteInPlace sway/config.c --replace '"swaybg"' '"${swaybg}/bin/swaybg"'
```

Comment:
We probably want to convert this to a patch to make sure that it never accidentally breaks in the future (though such problems are unlikely in this case). But we can keep it this way for now.

Reply:
Yeah, it's extremely unlikely to cause false-positives. If the location changes we could run into some problems, although the worst that will happen is the user not having a wallpaper.

I'm not entirely sure how we would do this with a patch, since it has to be dynamic? Are you suggesting to have the `substituteInPlace` applied to the patch, and then the patch applied to the file? What I'm doing here already is fairly common in nixpkgs IIRC.

Reply:
I agree that this is already a common pattern, I don't really see what would go wrong in this case. I do see a patch going wrong when there's an instance of "swaybg" that we _should_ patch but that doesn't get caught by the original patch.

Reply:
Yeah, that's what I'm saying.

Reply:
This could be possibly solved by just substituting on all files, although that naturally has a slightly higher risk and is less clean.

Reply:
> I'm not entirely sure how we would do this with a patch, since it has to be dynamic?

Oh, forgot about that. IIRC we have a dedicated function(?) for that (replacing patterns like `@swaybg@` in patches) but I forgot the name :o

Anyway, what I mainly wanted to say is that I might convert this into a patch in the future in case it will make things easier to maintain but for now it LGTM (but I realize now that I completely failed to properly state that in my initial nitpick, sorry about that).

Reply:
It's probably feasible to just use the Nix version of `substituteAll` directly in `patches`. I grepped through nixpkgs for instances of this and it seems to be a commonly used solution.

Reply:
Actually, this seems trivial. Give me a minute and I'll make the change myself.